sites/cheerpj/src/content/docs/13-tutorials/05-interoperability-tutorial.mdx

```diff
@@ -0,0 +1,159 @@
+Next, we create a JavaScript function that receives input from Java, displays it in the HTML document, and logs the input to the console.
+
+```js title="index.html"
+async function Java_com_example_Example_sendToHTML(lib, str) {
+	document.getElementById("javaOutput").innerText =
+		"JavaScript received: " + str;
+	console.log("Received input from Java: " + str);
+}
+```
+
+Here, the `Java_com_example_Example_sendToHTML` function in JavaScript acts as a native method, allowing a Java class to communicate with JavaScript.
+
+## Handling Multiple JavaScript Instances
+
+To establish persistent communication between Java and JavaScript, we’ll set up a long-running Java thread that enables JavaScript to access Java methods continuously. For example, using `cheerpjRunLibrary` to execute a Java class with native methods creates a separate process from `cheerpjRunJar`, which can prevent resources from being shared between instances.
+
+### Setting the Java Application Instance in JavaScript
+
+To establish consistent communication between Java and JavaScript instances, we use a long-running Java thread within the Java application.
+
+```java title="Example.java"
+public class Example {
+  public static native void nativeSetApplication(Example myApplication);
+
+  public static void main(String[] args) {
+        Example app = new Example();
+        new Thread(() -> {
+            nativeSetApplication(app);
+            System.out.println("Starting Thread");
+        }).start();
+    }
+}
+```
+
+We also use a native function to capture this thread on the JavaScript side, enabling persistent access to Java methods from JavaScript.
+
+```js title="index.html"
+async function Java_com_example_Example_nativeSetApplication(
+	lib,
+	myApplication
+) {
+	window.myApplication = myApplication;
+	console.log("Java application instance set on JavaScript side.");
+	return new Promise(() => {}); // Keeps the function from returning
+}
+```
+
+This function binds the Java application to a JavaScript object (`window.myApplication`) for easy access and prevents the function from returning, ensuring continuous accessibility.
+
+## Sending Data to Java from HTML
+
+Next, we create a JavaScript function that retrieves input from an HTML input box, sends it to the Java method, and displays the Java response in an alert.
+
+```js title="index.html"
+async function sendInputToJava() {
+	const inputText = document.getElementById("inputText").value;
+	const response = await window.myApplication.processInput(inputText);
+	window.alert(response);
+}
+```
+
+## Initializing CheerpJ with `cheerpjInit`
+
+To load CheerpJ, initialize it in your HTML with specific configuration for native methods:
+
+```js title="index.html"
+(async () => {
+	await cheerpjInit({
+		version: 8,
+		natives: {
+			Java_com_example_Example_sendToHTML,
+			Java_com_example_Example_nativeSetApplication,
+		},
+	});
+	cheerpjCreateDisplay(800, 600);
+	await cheerpjRunJar("/app/example.jar");
+})();
+```
```

Comment:
You should explain the `natives` init option a bit, and also explain the naming convention briefly.